TLM/TLM/Manager/Impl/ParkingRestrictionsManager.cs

```diff
@@ -25,15 +25,30 @@ public class ParkingRestrictionsManager
 
         public bool MayHaveParkingRestriction(ushort segmentId) {
             ref NetSegment segment = ref segmentId.ToSegment();
-            if ((segment.m_flags & NetSegment.Flags.Created) == NetSegment.Flags.None) {
-                return true;
+            if (!segment.IsValid()) {
+                return false;
             }
 
             ItemClass connectionClass = segment.Info.GetConnectionClass();
-
             return connectionClass.m_service == ItemClass.Service.Road && segment.Info.m_hasParkingSpaces;
         }
 
+        public bool MayHaveParkingRestriction(ushort segmentId, NetInfo.Direction finalDir) {
+            ref NetSegment segment = ref segmentId.ToSegment();
+            var dir = Shortcuts.RHT ? finalDir : NetInfo.InvertDirection(finalDir);
+            bool right = dir == NetInfo.Direction.Forward != segment.m_flags.IsFlagSet(NetSegment.Flags.Invert);
```

Comment:
That's a little bit confusing. I don't like double equality check with mixed types

Reply:
should I put parenthesis?
`bool != bool`  is xor
`bool == bool` is xnor